kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/internal/asset/provider/AssetOptions.java

```diff
@@ -103,11 +103,8 @@ private void addChannel(final Channel channel) {
      *
      * @param properties
      *            the properties to retrieve the channels from
-     * @throws NullPointerException
-     *             if the argument is null
      */
     private void checkChannelAvailability(final Map<String, Object> properties) {
-        requireNonNull(properties, message.propertiesNonNull());
```

Comment:
Those methods are private. The check is performed at the boundaries and we are assured that the map is not null. I completely agree in having contracts specified for public methods, since those define the interaction, and for these it's meaningful to check what is passed.
For private methods in a class it's a little bit an overkill, with an unneeded cost in terms of CPU time.

https://dev.eclipse.org/sonar/component/index#component=org.eclipse.kura%3Aorg.eclipse.kura.asset.provider%3Asrc%2Fmain%2Fjava%2Forg%2Feclipse%2Fkura%2Finternal%2Fasset%2Fprovider%2FAssetOptions.java&currentIssue=05c789f4-1d0e-4c9c-96dd-ebfa2bdcb17a